tests/test_monitor.py

```diff
@@ -0,0 +1,88 @@
+import random
+from time import sleep
+
+from sentry_sdk import Hub, start_transaction
+from sentry_sdk.transport import Transport
+
+
+class HealthyTestTransport(Transport):
+    def _send_event(self, event):
+        pass
+
+    def _send_envelope(self, envelope):
+        pass
+
+    def is_worker_full(self):
+        return False
+
+
+class UnhealthyTestTransport(HealthyTestTransport):
+    def is_worker_full(self):
+        return True
+
+
+def test_no_monitor_if_disabled(sentry_init):
+    sentry_init(transport=HealthyTestTransport())
+    assert Hub.current.client.monitor is None
+
+
+def test_monitor_if_enabled(sentry_init):
+    sentry_init(
+        transport=HealthyTestTransport(),
+        _experiments={"enable_backpressure_handling": True},
+    )
+
+    monitor = Hub.current.client.monitor
+    assert monitor is not None
+    assert monitor._thread is None
+
+    assert monitor.is_healthy() is True
+    assert monitor.downsample_factor() == 1
+    assert monitor._thread is not None
+    assert monitor._thread.name == "sentry.monitor"
+
+
+def test_monitor_unhealthy(sentry_init):
+    sentry_init(
+        transport=UnhealthyTestTransport(),
+        _experiments={"enable_backpressure_handling": True},
+    )
+
+    monitor = Hub.current.client.monitor
+    monitor.interval = 0.1
+
+    assert monitor.is_healthy() is True
+    sleep(0.1)  # wait for monitor to run
```

Comment:
if this thread stuff turns out to be flaky with timing, i'll change the logic later